docs/design/fabric-v2+/trusted_ledger.md

```diff
@@ -0,0 +1,162 @@
+        - for each item validate policy [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/policies/policy.go#L133)
+
+Lifecycle validation:
+
+
+Default validation:
+
+- Validating identities that signed the transaction
+    - read/write check?
+- Verifying the signatures of the endorsers on the transaction
+    - can endorse?
+- Ensuring the transaction satisfies the endorsement policies of the namespaces of the corresponding chaincodes.
+
+
+## Approach
+
+Restrict Fabric functionality:
+
+- keep Fabric validation changes in sync with TLCC
+- Reduces validation logic to a minimum
+    - minimizes catch-up 
+- Restrict the notion of endorsement policy; 
+- Simplify chaincode versioning (TODO)
+    - No chaincode updates for now
+- Support for default MSP only
+
+
+Any transactions/blocks with unsupported features are ignored/aborted
+
+Defined process to develop TLCC:
+- keep redundant validation code (in stock peer and TLCC) in sync
+- keep code relation traceable and changes trackable
+- consistent naming and code structure as much as possible
+- bonus: 
+    - automated code-changes notification; notifies whenever relevant Fabric code changes and might FPC must be updates.
+
+### Non-features
+
+- No state-based endorsement
+- No custom MSP (no idemix)
+- No support for custom endorsing/validation plugins for non-FPC and FPC chaincodes
+- Authentication and decorators
+- TODO check for more non-features in RFC
+
+## Design
+
+### Chaincode execution support
+    - secure channel
+    - validate proposal
+        - check org is "write"
+        - replay protection
+
+### Ledger maintainance
+
+Channel configuration:
+- Parse channel definition
+    - parse consensus definition
+- parse channel reader/writers
+- validate block signatures (Note that this check implies correctness as onfig blocks are validated by the ordering service ([see code](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/orderer/common/msgprocessor/standardchannel.go#L131)))
+- Maintain msp metadata for signature validation
+- parse Lifecycle policy 
+    - validate Lifecycle policy
+    We only support: a majority of Org.member (See [Link](https://hyperledger-fabric.readthedocs.io/en/release-2.2/chaincode_lifecycle.html#install-and-define-a-chaincode))
+
+Access control:
+- delegate this "reader/writer/admin" check to admins;
+- TLCC only verifies endorsement policies and thereby implicitly the "writers" check is performed by the endorsers
+
+MSP:
+- One Org - One MSP mapping - One root cert
+- Implement X509-based MSP
+    - Restrict:
+        - no intermediate
+        - no CRLs
+        - only support member role
+    - any certificate will match to role member
+
+Endorsing:
+- Phase1: Designated enclave only
+- Phase2: Any enclave that runs a particular FPC chaincode
+
+Versioning:
+- single autonomously monotonously increasing version number??
```

Comment:
monotonically increasing version numbers for us will be a bit a challenge with it being MRENCLAVE ... ;-)    [different story of course for ercc, though not sure whether for us the version really matters for ercc (compared to sequence numbers) as we look only at meta data changes and on that level version doesn't really have any effect? It is only relevant during the actual execution of the endorsement (which we delegate to normal peers for ercc)?